tensorboard/defs/defs.bzl

```diff
@@ -53,7 +55,7 @@ def tf_js_binary(name, compile, deps, visibility = None, **kwargs):
         **kwargs
     )
 
-    if compile:
+    if kwargs.get("dev_mode_only") == True:
```

Comment:
I’m confused about a bunch of things, so maybe I’m just misunderstanding
it all. It looks to me like, as written, `dev_mode_only` will be passed
through to `rollup_bundle` as `kwargs`, which is an error. I expected to
see something like:

```python
def tf_js_binary(
    name,
    compile,
    deps,
    visibility = None,
    dev_mode_only = False,
    **kwargs
):
    ...  # now `rollup_bundle(..., **kwargs)` is okay
```

But it also looks to me like the `dev_mode_only` condition here is
backward, and the real `tf_js_binary` target doesn’t actually set
`dev_mode_only = True`, so I’m not convinced that I understand…

In any case, agreed to start with a rollback.

Reply:
Sorry for the confusion, I made some mistakes and didn't actually upload the patch with the right changes.  You're totally correct that the `tf_js_binary` wasn't setting the flag, and didn't work.

Now that the offending PR was reverted, I've updated this PR to be a re-land of the original.  The original commit is the first one, and the 2nd commit in this branch contains new modifications.

The test sync cl has also been updated: cl/355900608